hack/scripts/validate_state.sh

```diff
@@ -17,7 +17,7 @@ do
     echo "Node internal ip: $node_ip"
     # Check pod count after restarting nodes, statefile does not exist after restart
     echo "checking whether the node has any pods deployed to it or not"
-    pod_count=$(kubectl get pods -A -o wide | grep "$node_name" -c)
+    pod_count=$(kubectl get pods -n kube-system -o wide | grep "$node_name" -c)
```

Comment:
I think we are deploying the pods to the `default` ns and validating their ips with the state file.

Reply:
If we want to check all namespaces. `-A` should have worked. Why are we restricting it to `kube-system`. Why is getting skipped ?
The idea of the check is to have something deployed after the restart as well so that the state file exists to validate.

Reply:
Previously the check was only looking in default, I changed it to `-A` a couple weeks back in a different PR.
Not restricting to kube-system permanently, I was testing and trying to see what was happening in the restart node check. 
In all the runs I've had where the clusters successfully created without kube-proxy, the restart node check has been failing to validate state, it gets stuck `trying to get the azure_endpoints` 
This seems like the validate state check is trying to validate when the nodes still don't have pods yet. How can we refine this pod count command to get the correct # pods on the node?

Reply:
We know that azure-endpoints won't exists when we restart the nodes if no new deployments happened after the restart. We can pass the variables `restart=true` in the pipeline yaml when we call the `validate_test.sh script` something like `./validate_test.sh --restart`
Then we can remove the pod check if condition altogether and add a limit to the for loop for getting the azure_endpoints.json( like try 5 times instead of forever). If we still fail to get the endpoints files, there are 2 cases to handle:
1. Restart node case - Check if we passed the `restart` variable, if so it is safe to assume that file didn't get created as no pods deployed to that node. Continue to next node.
2. Other cases - Fail in this case, as we are not in the `restart` phase and expect the azure_endpoints file to be present.

Let me know if you want to chat over this.

Reply:
update -- load test has been fixed with the necessary changes